crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -20,6 +21,77 @@ where
         unimplemented!()
     }
 
+    /// Finds a given trasaction by its hash.
+    pub(crate) async fn transaction_by_hash(
+        &self,
+        hash: H256,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let tx_by_hash = self.client().transaction_by_hash(hash)
+            .map_err(|_| EthApiError::EmptyRawTransactionData)?
+            .ok_or(EthApiError::EmptyRawTransactionData)?
+            .into_ecrecovered()
+            .ok_or(EthApiError::InvalidTransactionSignature)?;
+
+        Ok(Some(Transaction::from_recovered_with_block_context(
+            tx_by_hash,
+            // this is just stubbed out for now still need to fully implement
+            H256::default(),
+            u64::default(),
```

Comment:
@rakita how do we go from transaction to its block?

Reply:
We had this table but nuked it, right? Might be fine to bring it back.

Reply:
We had a different table but Oliver used same name when writing about it in telegram, we have `BlockNum-> TxCount` while @onbjerg  is proposing `TxCount -> BlockNum` it is different association, it is very elegant way to tie transaction number to block number.

Reply:
Are TxCount and the current TxNumber (that's already used in the tables) the same thing here?

Reply:
@sambukowski answered and clarified some things here: https://github.com/paradigmxyz/reth/issues/1699#issuecomment-1472256243
`TxNumber` is the correct type.